api/v1alpha1/virtualmachinepublishrequest_types.go

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// VirtualMachinePublishRequestConditionSourceValid is the Type for a
+	// VirtualMachinePublishRequest resource's status condition.
+	//
+	// The condition's status is set to true only when the information
+	// that describes the source side of the publication has been validated.
+	VirtualMachinePublishRequestConditionSourceValid = "SourceValid"
```

Comment:
Maybe we can move these conditions to https://github.com/vmware-tanzu/vm-operator-api/blob/master/api/v1alpha1/condition_consts.go?

Reply:
Could, but these are specific to this type and make more sense here IMO.